### ISSUE TITLE
docker deploy:

### ISSUE THREAD

**Issue by raarts** [docker/docker#30173]
**Description**

I am playing with docker deploy on docker 1.13-rc7. 

**Steps to reproduce the issue:**

issuing the following command repeatedly:

docker deploy -c product.yml product --with-registry-auth

**Describe the results you received:**

Error response from daemon: rpc error: code = 2 desc = changing network in service is not supported

**Describe the results you expected:**

Even if I was changing networks in a service, I would expect it to be handled gracefully.

**Additional information you deem important (e.g. issue happens only occasionally):**

The error seems to suggest it happens when I change networks in a service, but I really am not.

**Output of `docker version`:**

```
Client:
 Version:      1.13.0-rc7
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   48a9e53
 Built:        Fri Jan 13 06:37:02 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0-rc7
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   48a9e53
 Built:        Fri Jan 13 06:37:02 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 2
 Running: 1
 Paused: 0
 Stopped: 1
Images: 6
Server Version: 1.13.0-rc7
Storage Driver: overlay
 Backing Filesystem: extfs
 Supports d_type: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local-persist local
 Network: bridge host ipvlan macvlan null overlay
Swarm: active
 NodeID: ebj9sdrk4z0fl693z7g5ht5ix
 Is Manager: true
 ClusterID: mhdentu8t3yzti0z62r5h1hqh
 Managers: 3
 Nodes: 6
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 10.8.122.201
 Manager Addresses:
  10.8.122.201:2377
  10.8.127.73:2377
  10.8.128.73:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 2f7393a47307a16f8cee44a37b262e8b81021e3e
init version: 949e6fa
Kernel Version: 4.8.14-docker-1
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 3.857 GiB
Name: d0002.ams.tc.novantum.com
ID: INBZ:YDOR:C6RO:J7QU:CRRZ:MMVO:N3ZP:MEHO:LIFL:3UIW:7B4K:7N7M
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
 provider=generic
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

I *think* this started at rc7, but am not entirely sure. It can also be my deployment grew in complexity.

**Comment by thaJeztah** [comment]
ping @vdemeester @dnephin

**Comment by vdemeester** [comment]
There is a fix here : https://github.com/docker/docker/pull/29732 (and it's related to https://github.com/docker/docker/issues/29676).

**Comment by vdemeester** [comment]
Should be fixed by #29732
